app/src/main/java/com/rajatgoyal/puzzle15/ui/GameActivity.java

```diff
@@ -597,22 +508,56 @@ public boolean onTouch(View v, MotionEvent event) {
 
         @Override
         public void onSwipeRight() {
-            swipeHandler(view, SWIPE.RIGHT);
+            makeMove(0, -1);
```

Comment:
Should not to use magic numbers here and in other swipe handlers

Reply:
I didn't get your point. Please elaborate.

Reply:
By magic number, i meant unique values with unexplained meaning.. i.e 0, 1, -1 etc used in makeMove function call.

Reply:
Either make them constants/enums or evaluate them using known variables/constants by some means